lights-out.js

```diff
@@ -40,28 +39,60 @@ class LightsOut {
                 neighborArray.push([row, this.numCols - 1]);     
             }
         } else {
-            neighborArray.push([row, 0]);
-            neighborArray.push([row, col - 1]);
+            neighborArray.push([row, 0]), neighborArray.push([row, col - 1]);
         }
-
         return neighborArray;
     }
 
     registerClicks() {
-        for(let row = 0; row < this.numRows; row++) {
-            for(let col = 0; col < this.numCols; col++) {
+        for (let row = 0; row < this.numRows; row++) {
+            for (let col = 0; col < this.numCols; col++) {
                 const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
                 button.addEventListener('click', () => {
                     button.classList.toggle("on");
                     const neighbors = this.getNeighbors(row, col);
-                    
                     for (const neighbor of neighbors) {
                         const [r, c] = neighbor;
                         document.querySelector(`#${this.gameId} [data-row="${r}"][data-col="${c}"]`)
-                            .classList.toggle("on")
+                            .classList.toggle("on");
+                    }
+                    if (!button.textContent) {
+                        this.solve();
+                    } else {
+                        button.textContent = null;
                     }
                 });
             }
         }
     }
+
+    solve() {
+        for (let row = 0; row < this.numRows; row++) {
+            for (let col = 0; col < this.numCols; col++) {
+                document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`).textContent = null;
+            }
+        }
+        for (let row = 0; row < this.numRows; row++) {
+            for (let col = 0; col < this.numCols; col++) {
+                const button = document.querySelector(`#${this.gameId} [data-row="${row}"][data-col="${col}"]`);
+                const neighbors = this.getNeighbors(row, col);
+                if (button.classList.contains("on")) { 
+                    if (button.textContent) {
+                        button.textContent = null;
+                    } else {
+                        button.textContent = "Click me!";
+                    }

```

Comment:
I think you can remove this conditional by adding a new line after `const neighbors = ...`. The new line should be `neighbors.push([row, col]);`

This insight makes me wonder if we would have been better off to make `getNeighbors(row, col) {...}` include `[row, col]` in its return result. However, we are close to to finishing this project and we are past the good-idea cutoff point, so let's leave `getNeighbors(row, col) {...}` as is.

Reply:
This doesn't seem to be working, unless I am misunderstanding. I am deleting the conditional
```
if (button.classList.contains("on")) { 
                    if (button.textContent) {
                        button.textContent = null;
                    } else {
                        button.textContent = "Click me!";
                    }
```
and adding `neighbors.push([row, col]);` after `const neighbors = this.getNeighbors(row, col);`. 

Side note, there are actually two lines of `const neighbors = this.getNeighbors(row, col);`, so one is redundant.

Reply:
Let's discuss this issue during our meeting at 11am. For future reference, code reviews aren't the best place to collaboratively debug code. It's better to chat on other platforms, such as Discord. This way, it's easier for our teammates from the future to study the history of the code of this project.

Reply:
Gotcha